packages/react/src/components/Dialog/Dialog.base.tsx

```diff
@@ -102,11 +102,11 @@ export class DialogBase extends React.Component<IDialogProps, {}> {
     };
 
     let dialogDraggableClassName: string | undefined;
-    const dragOptions = modalProps?.dragOptions;
+    const dragOptions = { ...modalProps?.dragOptions };
 
     // if we are draggable, make sure we are using the correct
     // draggable classname and selectors
-    if (dragOptions && !dragOptions.dragHandleSelector) {
+    if (dragOptions && !modalProps?.dragOptions?.dragHandleSelector) {
```

Comment:
Why not use the dragOptions above? it seems like the `modalProps?.dragOptions` is the same as dragOptions.

Reply:
cleaned this up a bit, and fixed a bug as well